addons/rookupgrade/10to14/install.sh

```diff
@@ -0,0 +1,177 @@
+        # enabling autoscaling at this point doesn't scale down the number of PGs sufficiently in my testing - it will be enabled after installing 1.4.9
+        local osd_pools=
+        local non_osd_pools=
+        osd_pools="$(kubectl -n rook-ceph exec deploy/rook-ceph-tools -- ceph osd pool ls | grep rook-ceph-store)"
+        non_osd_pools="$(kubectl -n rook-ceph exec deploy/rook-ceph-tools -- ceph osd pool ls | grep -v rook-ceph-store)"
+        rookupgrade_10to14_scale_down_osd_pool_pg_num "$osd_pools" 16
+        rookupgrade_10to14_scale_down_osd_pool_pg_num "$non_osd_pools" 32
+        # log "Waiting for pool pgs to scale down"
+        # rookupgrade_10to14_osd_pool_wait_for_pg_num "$osd_pools" 16
+        # rookupgrade_10to14_osd_pool_wait_for_pg_num "$non_osd_pools" 32
+
+        "$DIR"/bin/kurl rook wait-for-health
+
+        logStep "Upgrading to Rook 1.1.9"
+
+        # first update rbac and other resources for 1.1
+        kubectl create --save-config -f "$upgrade_files_path/upgrade-from-v1.0-create.yaml" || true # resources may already be present
+        kubectl apply -f "$upgrade_files_path/upgrade-from-v1.0-apply.yaml"
+
+        # change the default osd pool size from 3 to 1
+        kubectl apply -f "$upgrade_files_path/rook-config-override.yaml"
+
+        kubectl delete crd --ignore-not-found volumesnapshotclasses.snapshot.storage.k8s.io volumesnapshotcontents.snapshot.storage.k8s.io volumesnapshots.snapshot.storage.k8s.io
+        kubectl -n rook-ceph set image deploy/rook-ceph-operator rook-ceph-operator=rook/ceph:v1.1.9
+        kubectl -n rook-ceph set image deploy/rook-ceph-tools rook-ceph-tools=rook/ceph:v1.1.9
+        echo "Waiting for Rook 1.1.9 to rollout throughout the cluster, this may take some time"
+        "$DIR"/bin/kurl rook wait-for-rook-version "v1.1.9"
+        # todo make sure that the RGW isn't getting stuck
+        echo "Rook 1.1.9 has been rolled out throughout the cluster"
+
+        echo "Upgrading CRDs to Rook 1.1"
+        kubectl apply -f "$upgrade_files_path/upgrade-from-v1.0-crds.yaml"
+
+        semverCompare "$(current_ceph_version)" "14.2.5"
+        if [ "$SEMVER_COMPARE_RESULT" != "1" ]; then
+            echo "Upgrading to Ceph 14.2.5"
+
+            kubectl -n rook-ceph patch CephCluster rook-ceph --type=merge -p '{"spec": {"cephVersion": {"image": "ceph/ceph:v14.2.5-20191210"}}}'
+            kubectl patch deployment -n rook-ceph csi-rbdplugin-provisioner -p '{"spec": {"template": {"spec":{"containers":[{"name":"csi-snapshotter","imagePullPolicy":"IfNotPresent"}]}}}}'
+            "$DIR"/bin/kurl rook wait-for-ceph-version "14.2.5"
+
+            "$DIR"/bin/kurl rook wait-for-health
+
+            echo "Upgraded to Ceph 14.2.5 successfully"
+        fi
+
+        logSuccess "Upgraded to Rook 1.1.9 successfully"
+    fi
+
+    if [ "$(rook_upgrade_compare_rook_versions "$from_version" "1.2")" != "1" ]; then
+        logStep "Upgrading to Rook 1.2.7"
```

Comment:
I think if we're going to do this, we should have a wait-for-health call at the start of each function after the log line

that way we don't further fuck up broken rook

```suggestion
    if [ "$(rook_upgrade_compare_rook_versions "$from_version" "1.2")" != "1" ]; then
        logStep "Upgrading to Rook 1.2.7"
        "$DIR"/bin/kurl rook wait-for-health
```